lmdeploy/messages.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+
+from dataclasses import dataclass
+from typing import List
+
+from .tokenizer import Tokenizer
+
+
+@dataclass
+class GenerationConfig:
+    """generation parameters used by inference engines."""
+
+    n: int = 1  # How many chat completion choices to generate for each input message. # noqa E501
+    max_new_tokens: int = 512  # The maximum number of tokens that can be generated in the chat completion. # noqa E501
+    top_p: float = 1.0  # An alternative to sampling with temperature, called nucleus sampling, where the model considers the results of the tokens with top_p probability mass # noqa E501
+    top_k: int = 1  # An alternative to sampling with temperature, where the model considers the top_k tokens with the highest probability # noqa E501
+    temperature: float = 0.8  # sampling temperature
+    repetition_penalty: float = 1.0  # penalty to prevent the model from generating repeated words or phrases. A value larger than 1 discourages repetition. # noqa E501
+    ignore_eos: bool = False  # indicator to ignore the eos_token_id or not
+    random_seed: int = None  # seed used when sampling a token
+    stop_words: List[str] = None  # words that stop generating further tokens
+    bad_words: List[str] = None  # words that the engine will never generate
+
+
+@dataclass
+class EngineGenerationConfig:
```

Comment:
Could we inherit from `GenerationConfig` to avoid multiple definitions of the same arguments?
```
@dataclass
class EngineGenerationConfig(GenerationConfig):
    stop_words: List[int] = None
    bad_words: List[int] = None
   ...
```

Reply:
Does `stop_words` override the base classs's `stop_words`?

Reply:
Yes.
```
import inspect
print(inspect.getfullargspec(EngineGenerationConfig).annotations['stop_words'])
## typing.List[int]
```

Reply:
Fixed